crates/wash-cli/src/common/registry_cmd.rs

```diff
@@ -162,6 +194,58 @@ pub async fn handle_command(
     }
 }
 
+fn resolve_artifact_ref(url: &str, registry: &str) -> Result<Reference> {
+    let image: Reference = url
+        .trim()
+        .to_ascii_lowercase()
+        .parse()
+        .context("failed to parse artifact url into oci image reference")?;
+
+    if url.trim() == image.whole() {
```

Comment:
I'm not very familiar with the `oci_distribution` crate. When will this not be true?

Reply:
based on our use case and the implementation of `image.whole()` from the `oci_distribution` crate, that predicate will be false when the user provides a repository path rather than a full artifact url.

Reply:
will mark this as resolved but you follow up on this if you think there should be more deliberation on this